Squeezer/src/main/res/layout/squeezer_remote_control_next_button.xml

```diff
@@ -0,0 +1,12 @@
+<ImageButton xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/squeezerRemote_next"
+    style="?android:attr/buttonBarButtonStyle"
+    android:layout_width="@dimen/squeezerremote_button_size"
+    android:layout_height="@dimen/squeezerremote_button_size"
+    android:layout_marginLeft="-4dip"
+    android:layout_marginTop="-5dip"
+    android:layout_marginRight="-4dip"
+    android:layout_marginBottom="-5dip"
+    android:contentDescription="@string/remote_nextDescription"
+    android:padding="0dp"
+    android:src="@drawable/ic_action_next" />
```

Comment:
These vector drawables seem to have an incompatibility with earlier android versions. I tried using the suggested appcompat versions but they didn't work at all. I'm open to ideas.

Reply:
To use vector drawables when using VectorDrawableCompat, you need to use app:srcCompat (hint try to build from command line with "./gradlew build", lint will tell you what to fix).
However vector drawables in RemoteViews doesn't work with earlier Android versions. See for example here [stackoverflow](https://stackoverflow.com/questions/35633410/appcompat-23-2-use-vectordrawablecompat-with-remoteviews-appwidget-on-api21).
I solved this for notifications in layout: notification_player_normal and this method in SqueezeService:
```
    private Bitmap vectorToBitmap(@DrawableRes int vectorResource) {
        Drawable drawable = AppCompatResources.getDrawable(this, vectorResource);
        Bitmap b = Bitmap.createBitmap(drawable.getIntrinsicWidth(), drawable.getIntrinsicHeight(), Bitmap.Config.ARGB_8888);
        Canvas c = new Canvas(b);
        drawable.setBounds(0, 0, c.getWidth(), c.getHeight());
        drawable.draw(c);
        return b;
    }
```

Reply:
Is there a good way to do this without moving what-should-probably-be-in-the-xml into the Java code?

Reply:
Not that I know of. Setting your vector drawable resources inside your RemoteViews layout file does not work (neither android:src nor app:srcCompat). You'll have to set them programmatically.
I.e remove android:src="@drawable/ic_action_next" from squeezer_remote_control_next_button.xml.
and in SqueezerRemoteControl do:
```
views.setImageViewBitmap(R.id.squeezerRemote_next, vectorToBitmap(R.drawable.ic_action_next));
```
if the corresponding button is displayed